docs/advanced_topics/headless.md

```diff
@@ -0,0 +1,190 @@
+(headless)=
+
+# Headless support
+
+Wagtail has good support for headless sites, but there are some issues developers should take into account when using Wagtail as a headless CMS.
+This page covers most topics related to headless sites, and tries to identify where you might run into issues using 🟢 (good support), 🟡 (workarounds needed or incomplete support) and 🔴 (lacking support).
```

Comment:
question: is there a way to make these more accessible (e.g. for folks with colour blindness)?

Reply:
Here's a suggestion - feel free to reject, of course: 

To help with different perceptions of colour _and_ also usage of screen readers, we could move the emoji till the line after the title, and also folliowing it with a two-word summary term (e.g. Good support / Incomplete support / Lacking support) would 

e.g. 

```
# REST
🟢  Good support
REST (or representational state transfer) was introduced...
````

Sounds like

> R E S T 
> Green circle good support
> R E S T (or representational state transfer) was introduced ...

rather than 

> Green circle R E S T
> R E S T (or representational state transfer) was introduced ...

But, it would make the left-side navigation in the rendered page less colourful/visually scannable for those who don't have perception challenges.

Reply:
Here are renderings of different options:

----
The current version in this PR:
<img width="1253" alt="Screenshot 2024-06-13 at 15 50 26" src="https://github.com/wagtail/wagtail/assets/143557/d267f473-68a9-4155-a830-a0715b0f992c">

----
The emojis changed as discussed above:

<img width="1240" alt="Screenshot 2024-06-13 at 15 50 33" src="https://github.com/wagtail/wagtail/assets/143557/b5b7da0e-89ab-4b31-9c09-a7c8a1426e53">

----
Emojis moved to the end of heading

<img width="1245" alt="Screenshot 2024-06-13 at 15 50 41" src="https://github.com/wagtail/wagtail/assets/143557/4094445d-f82d-4c42-a334-a0cd529b7295">

----
Emojis below heading, with a clarification (Good, incomplete or lacking support)

<img width="1231" alt="Screenshot 2024-06-13 at 15 50 48" src="https://github.com/wagtail/wagtail/assets/143557/34a6b3dc-fae7-4c60-8e24-d4ca9b00deb5">

Reply:
I think the second solution shown above (emoji, with no extra description) could be a good balance between things - they have a better degree of meaning than just coloured circles, even if emoji interpretation can vary across cultures. Their broader meaning is covered in the intro on the page too, which def helps.

Reply:
Thanks @stevejalim - I've committed that change.